src/modules/xcb/xcbkeyboard.cpp

```diff
@@ -433,7 +433,7 @@ void XCBKeyboard::setRMLVOToServer(const std::string &rule,
     do {                                                                       \
         *data = FIELD##Len;                                                    \
         data += 1;                                                             \
-        if (FIELD##Len) {                                                      \
+        if (FIELD##Len && rnames.FIELD) {                                      \
```

Comment:
this is false positive since the null check is done above. Non-Zero length means not null.